core/controllers/contributor_dashboard.py

```diff
@@ -524,3 +524,40 @@ def get(self):
             'topic_names': topic_names
         }
         self.render_json(self.values)
+
+
+class TranslationPreferenceHandler(base.BaseHandler):
+    """Provides preffered translation language for the
+    Contribution dashboard page.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'POST': {
+            'language_code': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        }
+    }
+
+    @acl_decorators.open_access
+    def get(self):
+        """Handles GET requests."""
+        user_settings = user_services.get_user_settings(self.user_id)
+        self.values.update({
+            'preferred_text_language_code': (
+                user_settings.preferred_text_language_code)
+        })
+        self.render_json(self.values['preferred_text_language_code'])
```

Comment:
We typically return the dictionary of values. See above for examples.

Reply:
Updated this.